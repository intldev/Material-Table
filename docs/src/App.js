import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import MaterialTable from 'material-table'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selection: true,
      filtering: true,
      search: true,
      showColumnsButton: true,
      showPrintButton: true,
      paging: true,
      title: 'Title',
      showActions: true,
      toolbar: true,
      maxWidth: '100',
      zCount: 1,
      columns: [
        {
          title: 'Adı',
          field: 'name'          
        },
        {
          title: 'Soyadı',
          field: 'surname'          
        },
        {
          title: 'Doğum Yılı',
          field: 'birthYear',
          isNumeric: true
        },
        {
          title: 'Doğum Yeri',
          field: 'birthCity',
          lookup: {34: 'İstanbul', 63: 'Şanlıurfa'}
        }
      ],
      data: [
        {name: 'Mehmet', surname: 'Baran', birthYear: 1987, birthCity: 63},
        {name: 'Gülcan', surname: 'Baran', birthYear: 1989, birthCity: 34},
      ],
      actions: [
        {
          icon: 'save',
          tooltip: 'Save',
          action: (event, data) => {
            if (Array.isArray(data)) {
              alert("Save : " + data.length);
            } 
            else {
              alert("Save : " + data.name);
            }
          }
        },
        {
          icon: 'delete',
          tooltip: 'Delete',
          action: (event, data) => {
            if (Array.isArray(data)) {
              alert("Delete : " + data.length);
            } 
            else {
              alert("Delete : " + data.name);
            }
          }
        }
      ]
    }

    
    
  }  

  render() {
    const inputStyle={marginLeft: 40};
    const data = [...this.state.data]

    for(let i = 1; i <= this.state.zCount; i++) {
      data.push({name: i + '.Zerya Betül', surname: 'Baran', birthYear: 2017 + i, birthCity: i})
    }

    return (
      <div className="App" style={{backgroundColor: '#4aacc71a', height: '100%', padding: 50}}>
        Grid Demo
        <p className="App-intro">          
          <div style={{margin: 30, marginBottom: 100}}>
            <input style={inputStyle} type="checkbox" checked={this.state.filtering} onChange={(event) => this.setState({filtering: event.target.checked})} /> Filtering                          
            <input style={inputStyle} type="checkbox" checked={this.state.selection} onChange={(event) => this.setState({selection: event.target.checked})} /> Selection                          
            <input style={inputStyle} type="checkbox" checked={this.state.search} onChange={(event) => this.setState({search: event.target.checked})} /> Search                          
            <input style={inputStyle} type="checkbox" checked={this.state.showColumnsButton} onChange={(event) => this.setState({showColumnsButton: event.target.checked})} /> ShowColumnsButton                          
            <input style={inputStyle} type="checkbox" checked={this.state.showPrintButton} onChange={(event) => this.setState({showPrintButton: event.target.checked})} /> showPrintButton                          
            <input style={inputStyle} type="checkbox" checked={this.state.paging} onChange={(event) => this.setState({paging: event.target.checked})} /> Paging                          
            <input style={inputStyle} type="checkbox" checked={this.state.showActions} onChange={(event) => this.setState({showActions: event.target.checked})} /> Actions                          
            <input style={inputStyle} type="checkbox" checked={this.state.toolbar} onChange={(event) => this.setState({toolbar: event.target.checked})} /> Toolbar                          
            <span style={inputStyle}>Title</span><input type="text" value={this.state.title} onChange={(event) => this.setState({title: event.target.value})} /> 
            <span style={inputStyle}>Max Width</span><input type="text" value={this.state.maxWidth} onChange={(event) => this.setState({maxWidth: event.target.value})} /> 
            <span style={inputStyle}>Data Count</span><input type="text" value={this.state.zCount} onChange={(event) => this.setState({zCount: event.target.value})} /> 
          </div>

          <div style={{maxWidth: this.state.maxWidth.toString() + '%'}}>
            <MaterialTable
              actions={this.state.showActions && this.state.actions}
              columns={this.state.columns}
              data={data}
              options={{
                filtering: this.state.filtering,
                selection: this.state.selection,
                paging: this.state.paging,
                toolbar: this.state.toolbar && {
                  title: this.state.title,  
                  search: this.state.search,
                  showColumnsButton: this.state.showColumnsButton,
                  showPrintButton: this.state.showPrintButton
                }
              }}
            />
          </div>
        </p>
      </div>
    );
  }
}

export default App;
