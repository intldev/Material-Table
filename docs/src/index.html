<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">
  <meta name="theme-color" content="#000000">
  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500">
  <style>
    body {
      height: 100%;
      background-color: #4aacc71a;
      margin: 0;
      padding: 0;
      font-family: sans-serif;
    }
    @media only screen and (max-width: 600px) {
      .content {
        padding: 10px 0px 10px 0px;
      }
    }
    @media (min-width: 601px) {
      .content {
        padding: 10px 50px 10px 50px;
      }
    }
  </style>
  <title>material-table demo</title>
</head>
<body>
  <div id="react-div"/>
</body>
</html>
