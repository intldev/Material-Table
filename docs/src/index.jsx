import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import MaterialTable from 'material-table'
import { 
  Card, CardContent, Switch,
  Grid, TextField, AppBar, 
  Toolbar,
  List, ListItem, ListItemText, 
  ListItemSecondaryAction, Typography
} from '@material-ui/core'

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selection: true,
      filtering: true,
      search: true,
      showColumnsButton: true,
      showPrintButton: true,
      paging: true,
      title: 'Title',
      showActions: true,
      toolbar: true,
      maxWidth: '100',
      dataCount: 100,
      columns: [
        {
          title: 'Adı',
          field: 'name'          
        },
        {
          title: 'Soyadı',
          field: 'surname'          
        },
        {
          title: 'Doğum Yılı',
          field: 'birthYear',
          isNumeric: true
        },
        {
          title: 'Doğum Yeri',
          field: 'birthCity',
          lookup: {34: 'İstanbul', 63: 'Şanlıurfa'}
        }
      ],
      data: [
        {name: 'Mehmet', surname: 'Baran', birthYear: 1987, birthCity: 63},
      ],
      actions: [
        {
          icon: 'save',
          tooltip: 'Save',
          onClick: (event, data) => {
            if (Array.isArray(data)) {
              alert("Save : " + data.length);
            } 
            else {
              alert("Save : " + data.name);
            }
          }
        },
        {
          icon: 'delete',
          tooltip: 'Delete',
          onClick: (event, data) => {
            if (Array.isArray(data)) {
              alert("Delete : " + data.length);
            } 
            else {
              alert("Delete : " + data.name);
            }
          }
        }       
      ]
    }

    
    
  }  

  render() {
    const listStyle={backgroundColor: '#4aacc71a', margin: 2}
    const data = [...this.state.data]

    for(let i = 1; i < this.state.dataCount; i++) {
      data.push({name: 'Name ' + i, surname: 'Surname ' + i, birthYear: 2018 + i, birthCity: i})
    }

    return (
      <div>
        <AppBar position="static">
          <Toolbar>
            <Typography variant="title" color="inherit">
              material-table demo
            </Typography>
          </Toolbar>
        </AppBar>
        <div class="content">
          <div style={{padding:8}}>
            <Card style={{padding: 10}}>
              <Typography variant="title" style={{textAlign: 'left'}}>Options</Typography>
              <CardContent style={{padding:5}}>
                <Grid container>
                  <Grid item xs={12} sm={4} md={4} lg={3}>
                    <List style={listStyle} dense>
                      <ListItem>
                        <ListItemText primary="Title" />
                        <ListItemSecondaryAction>
                          <TextField
                            onChange={(event) => this.setState({title: event.target.value})}
                            value={this.state.title}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Data Count" />
                        <ListItemSecondaryAction>
                          <TextField
                            onChange={(event) => this.setState({dataCount: event.target.value})}
                            value={this.state.dataCount}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Max Width (%)" />
                        <ListItemSecondaryAction>
                          <TextField
                            onChange={(event) => this.setState({maxWidth: event.target.value})}
                            value={this.state.maxWidth}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      
                    </List>
                  </Grid>
                  <Grid item xs={12} sm={4} md={4} lg={3}>
                    <List style={listStyle} dense>
                      <ListItem>
                        <ListItemText primary="Filtering" />
                          <ListItemSecondaryAction>
                            <Switch
                              onChange={(event) => this.setState({filtering: event.target.checked})}
                              checked={this.state.filtering}
                            />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Selection" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({selection: event.target.checked})}
                            checked={this.state.selection}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Paging" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({paging: event.target.checked})}
                            checked={this.state.paging}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>                      
                    </List>
                  </Grid>
                  <Grid item xs={12} sm={4} md={4} lg={3}>
                    <List style={listStyle} dense>
                      <ListItem>
                        <ListItemText primary="Show Toolbar" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({toolbar: event.target.checked})}
                            checked={this.state.toolbar}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Show Columns Button" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({showColumnsButton: event.target.checked})}
                            checked={this.state.showColumnsButton}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Show Print Button" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({showPrintButton: event.target.checked})}
                            checked={this.state.showPrintButton}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                    </List>
                  </Grid>
                  <Grid item xs={12} sm={4} md={4} lg={3}>
                    <List style={listStyle} dense>
                      <ListItem>
                        <ListItemText primary="Search" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({search: event.target.checked})}
                            checked={this.state.search}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                      <ListItem>
                        <ListItemText primary="Add Actions" />
                        <ListItemSecondaryAction>
                          <Switch
                            onChange={(event) => this.setState({showActions: event.target.checked})}
                            checked={this.state.showActions}
                          />
                        </ListItemSecondaryAction>
                      </ListItem>
                    </List>
                  </Grid>
                </Grid>
              </CardContent>
            </Card>
          </div>

          <div style={{maxWidth: this.state.maxWidth.toString() + '%'}}>
            <MaterialTable
              actions={this.state.showActions && this.state.actions}
              columns={this.state.columns}
              data={data}
              title={this.state.title}
              options={{
                filtering: this.state.filtering,
                selection: this.state.selection,
                paging: this.state.paging,
                toolbar: this.state.toolbar,
                search: this.state.search,
                columnsButton: this.state.showColumnsButton,
                printButton: this.state.showPrintButton                
              }}
            />
          </div>
        </div>
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('react-div'));


