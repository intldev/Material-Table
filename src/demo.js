import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import MaterialTable from './material-table';

class App extends Component {
  state = {
    selectedCount: 0,
    data: [
      { id: 1, name: 'ax', surname: 'Baran', birthYear: 1987, birthCity: 63, sex: 'Male', type: 'adult' },
      { id: 2, name: 'bx', surname: 'Baran', birthYear: 1987, birthCity: 34, sex: 'Female', type: 'adult', parentId: 1 },
      { id: 3, name: 'cx', surname: 'Baran', birthYear: 1987, birthCity: 34, sex: 'Female', type: 'child', parentId: 1 },
      { id: 4, name: 'dx', surname: 'Baran', birthYear: 1987, birthCity: 34, sex: 'Female', type: 'child', parentId: 3 },
      { id: 5, name: 'ex', surname: 'Baran', birthYear: 1987, birthCity: 34, sex: 'Female', type: 'child' },
      { id: 6, name: 'fx', surname: 'Baran', birthYear: 2012, birthCity: 34, sex: 'Female', type: 'child', parentId: 5 },
    ],
    columns: [
      { title: 'Adı', field: 'name' },
      { title: 'Soyadı', field: 'surname', export: false },
      { title: 'Cinsiyet', field: 'sex', disableClick: true },
      { title: 'Tipi', field: 'type', removable: false },
      {
        title: 'Doğum Yılı', field: 'birthYear', type: 'numeric', cellStyle: data => {
          if (data > 2000) {
            return {
              color: 'red',
            };
          }
        },
      },
      { title: 'Doğum Yeri', field: 'birthCity', lookup: { 34: 'İstanbul', 63: 'Şanlıurfa' } },
    ],
    remoteColumns: [
      { title: 'Avatar', field: 'avatar', render: rowData => <img style={{ height: 36, borderRadius: '50%' }} src={rowData.avatar} /> },
      { title: 'Id', field: 'id' },
      { title: 'First Name', field: 'first_name' },
      { title: 'Last Name', field: 'last_name' },
    ]
  }

  render() {
    return (
      <div style={{ maxWidth: '100%' }}>
        <MaterialTable
          columns={this.state.columns}
          data={this.state.data}
          title="Demo Title"
          options={{
            filtering: true,
            columnsButton: true
          }}
        />
        <button
          onClick={() => this.setState({ selectedCount: this.state.selectedCount + 1 })}
        >
          {this.state.selectedCount}
        </button>
        <MaterialTable
          // columns={this.state.columns}
          // data={this.state.data}
          columns={this.state.remoteColumns}
          data={(query) => new Promise((resolve, reject) => {
            let url = "https://reqres.in/api/users?";
            url += "per_page=" + query.pageSize;
            url += "&page=" + (query.page + 1);

            /*global fetch:false*/
            fetch(url).then(response => response.json()).then(result => {
              resolve({
                data: result.data,
                page: result.page - 1,
                totalCount: result.total
              });
            });
          })}
          title="Demo Title"
        />



      </div>
    );
  }
}

ReactDOM.render(
  <App />,
  document.getElementById('app')
);

module.hot.accept();
