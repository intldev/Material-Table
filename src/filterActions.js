
export default options => a => {
    if (options.selection) {
        return a.isRowAction;
    } else {
        return (!a.isFreeAction && !a.isSelectedAction) || (a.isRowAction);
    }
};