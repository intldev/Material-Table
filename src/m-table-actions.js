/* eslint-disable no-unused-vars */
import * as React from 'react';
import PropTypes from 'prop-types';
import { Icon, IconButton, Tooltip } from '@material-ui/core';
/* eslint-enable no-unused-vars */

class MTableActions extends React.Component {
  renderButton(action, index) {
    if (typeof action === 'function') {
      action = action(this.props.data);
      if (!action) {
        return null;
      }
    }

    const button = (
      <IconButton
        key={action.icon + '' + index}
        color="inherit"
        disabled={action.disabled}
        onClick={(event) => action.onClick && action.onClick(event, this.props.data)}
      >        
        {typeof action.icon === "string" ?
          <Icon {...action.iconProps}>{action.icon}</Icon>
          :
          <action.icon {...action.iconProps}/>
        } 
      </IconButton>
    );

    if (action.tooltip && !action.disabled) {
      return <Tooltip title={action.tooltip} key={action.tooltip + '' + index}>{button}</Tooltip>;
    } else {
      return button;
    }
  }

  render() {
    if (this.props.actions) {
      return this.props.actions.map((action, index) => (this.renderButton(action, index)));
    }

    return null;
  }
}

MTableActions.defaultProps = {
  actions: [],
  data: {}
};

MTableActions.propTypes = {
  actions: PropTypes.array.isRequired,
  data: PropTypes.oneOfType([PropTypes.object, PropTypes.arrayOf(PropTypes.object)])
};

export default MTableActions;
