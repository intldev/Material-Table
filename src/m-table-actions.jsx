import * as React from 'react'
import {
  Icon, IconButton, Tooltip
} from '@material-ui/core'

class MTableActions extends React.Component {  

  renderButton(action) {
    const button = (
      <IconButton
        onClick={(event) => action.action && action.action(event, this.props.data)}
      >
        <Icon>{action.icon}</Icon>
      </IconButton>
    );

    if(action.tooltip) {
      return <Tooltip title={action.tooltip}>{button}</Tooltip>
    }
    else {
      return button;
    }
  }

  render() {
    if(this.props.actions) {
      return this.props.actions.map(action => (this.renderButton(action)));
    }

    return null
  }
}

export default MTableActions
