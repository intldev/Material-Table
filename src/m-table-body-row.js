/* eslint-disable no-unused-vars */
import * as React from 'react';
import PropTypes from 'prop-types';
import MTableCell from './m-table-cell';
import MTableActions from './m-table-actions';
import filterActions from './filterActions';
import { Checkbox, TableRow, TableCell } from '@material-ui/core';
/* eslint-enable no-unused-vars */


export default class MTableBodyRow extends React.Component {
  renderColumns() {
    const mapArr = this.props.columns.filter(columnDef => { return !columnDef.hidden })
      .map((columnDef) => {
        const value = this.props.getFieldValue(this.props.data, columnDef);
        return <MTableCell columnDef={columnDef} value={value} key={columnDef.tableData.id} rowData={this.props.data} />;
      });
    return mapArr;
  }

  renderActions() {
    return (
      <TableCell style={{ paddingTop: 0, paddingBottom: 0 }} key="key-actions-column">
        <div style={{ display: 'flex' }}>
          <MTableActions data={this.props.data} actions={this.props.actions.filter(filterActions(this.props.options))} />
        </div>
      </TableCell>
    );
  }
  renderSelectionColumn() {
    return (
      <TableCell padding="checkbox" key="key-selection-column">
        <Checkbox
          checked={this.props.data.tableData.checked === true}
          value={`${this.props.data.tableData.id}`}
          onChange={this.props.onRowSelected}
        />
      </TableCell>
    );
  }
  render() {
    const columns = this.renderColumns();
    if (this.props.options.selection) {
      columns.splice(0, 0, this.renderSelectionColumn());
    }
    if (this.props.actions &&
      this.props.actions.filter(filterActions(this.props.options)).length > 0) {
      if (this.props.options.actionsColumnIndex === -1) {
        columns.push(this.renderActions());
      } else if (this.props.options.actionsColumnIndex >= 0) {
        let endPos = 0;
        if (this.props.options.selection) {
          endPos = 1;
        }
        columns.splice(this.props.options.actionsColumnIndex + endPos, 0, this.renderActions());
      }
    }
    return (
      <TableRow selected={this.props.index % 2 === 0}>
        {columns}
      </TableRow>
    );
  }
}

MTableBodyRow.defaultProps = {
  actions: [],
  index: 0,
  data: {},
  options: {}
};

MTableBodyRow.propTypes = {
  actions: PropTypes.array,
  index: PropTypes.number.isRequired,
  data: PropTypes.object.isRequired,
  options: PropTypes.object.isRequired,
  onRowSelected: PropTypes.func,
  getFieldValue: PropTypes.func.isRequired,
  columns: PropTypes.array,
};
