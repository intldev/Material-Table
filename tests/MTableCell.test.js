import React from 'react';
import MTableCell from '../src/components/m-table-cell';
import renderer from 'react-test-renderer';
import { defaultProps } from '../src/default-props';

const passedDefaultProps = { icons: defaultProps.icons };
const testDate = new Date('1991-09-23T10:00:00.000Z');

test.each([
	['boolean', true, { type: 'boolean' }, passedDefaultProps],
	['date', testDate, { type: 'date' }],
	['datetime', testDate, { type: 'datetime' }],
	['time', testDate, { type: 'time' }],
	[
		'currency',
		100,
		{
			type: 'currency',
			currencyCode: 'GBP',
			currencySetting: { locale: 'en-UK' },
			locale: 'en-UK',
			minimumFractionDigits: 2,
			maximumFractionDigits: 2,
		},
	],
	['generic', 'abc'],
	['empty', undefined, { emptyValue: '' }],
])('MTableCell can render %p cell', (_name, value, columnDef = {}, props = {}) => {
	const tree = renderer.create(<MTableCell columnDef={columnDef} value={value} {...props} />);
	expect(tree.toJSON()).toMatchSnapshot();
});
