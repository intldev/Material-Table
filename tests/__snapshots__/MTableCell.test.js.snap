// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MTableCell can render "boolean" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value={true}
>
  <span
    aria-hidden={true}
    className="material-icons MuiIcon-root"
    style={
      Object {
        "textAlign": "left",
        "verticalAlign": "middle",
        "width": 48,
      }
    }
  >
    check
  </span>
</td>
`;

exports[`MTableCell can render "currency" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value={100}
>
  $100.00
</td>
`;

exports[`MTableCell can render "date" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value={1991-09-23T10:00:00.000Z}
>
  9/23/1991
</td>
`;

exports[`MTableCell can render "datetime" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value={1991-09-23T10:00:00.000Z}
>
  9/23/1991, 12:00:00 PM
</td>
`;

exports[`MTableCell can render "empty" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
>
  
</td>
`;

exports[`MTableCell can render "generic" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value="abc"
>
  abc
</td>
`;

exports[`MTableCell can render "time" cell 1`] = `
<td
  aria-sort={null}
  className="MuiTableCell-root MuiTableCell-alignLeft"
  onClick={[Function]}
  style={Object {}}
  value={1991-09-23T10:00:00.000Z}
>
  12:00:00 PM
</td>
`;
